air/src/execution_step/boxed_value/stream.rs

```diff
@@ -34,35 +38,51 @@ pub struct Stream {
     /// obtained values from a current_data that were not present in prev_data becomes a new generation.
     values: Vec<Vec<ValueAggregate>>,
 
+    /// Count of values from previous data.
+    previous_gens_count: usize,
+
     /// This map is intended to support canonicalized stream creation, such streams has
     /// corresponding value positions in a data and this field are used to create such streams.
     values_by_pos: HashMap<TracePos, StreamValueLocation>,
 }
 
 impl Stream {
-    pub(crate) fn from_generations_count(count: usize) -> Self {
+    pub(crate) fn from_generations_count(previous_count: usize, current_count: usize) -> Self {
+        let last_generation_count = 1;
+        // TODO: check for overflow
```

Comment:
I think it should be implemented before the tomorrow event.  Even if it just panics.